metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py

```diff
@@ -163,33 +164,31 @@ def _should_retry(exc: BaseException) -> bool:
             return True
 
         with self.report.list_projects:
-            try:
-                # Bigquery API has limit in calling project.list request i.e. 2 request per second.
-                # https://cloud.google.com/bigquery/quotas#api_request_quotas
-                # Whenever this limit reached an exception occur with msg
-                # 'Quota exceeded: Your user exceeded quota for concurrent project.lists requests.'
-                # Hence, added the api request retry of 15 min.
-                # We already tried adding rate_limit externally, proving max_result and page_size
-                # to restrict the request calls inside list_project but issue still occured.
-                projects_iterator = self.bq_client.list_projects(
-                    retry=retry.Retry(
-                        predicate=_should_retry, initial=10, maximum=180, timeout=900
-                    )
+            self.report.num_list_projects_api_requests += 1
+            # Bigquery API has limit in calling project.list request i.e. 2 request per second.
+            # https://cloud.google.com/bigquery/quotas#api_request_quotas
+            # Whenever this limit reached an exception occur with msg
+            # 'Quota exceeded: Your user exceeded quota for concurrent project.lists requests.'
+            # Hence, added the api request retry of 15 min.
+            # We already tried adding rate_limit externally, proving max_result and page_size
+            # to restrict the request calls inside list_project but issue still occured.
+            projects_iterator = self.bq_client.list_projects(
+                retry=retry.Retry(
+                    predicate=_should_retry, initial=10, maximum=180, timeout=900
                 )
-                projects: List[BigqueryProject] = [
-                    BigqueryProject(id=p.project_id, name=p.friendly_name)
-                    for p in projects_iterator
-                ]
-                self.report.num_list_projects = len(projects)
-                return projects
-            except Exception as e:
```

Comment:
is this exception handled elsewhere?

Reply:
Yes, this is moved up to caller of this method , where caller catches and reports error along with failure.